### PR TITLE
Feature/webhooks (UrbanAirship & Sendgrid)

### PR DESCRIPTION
**Pull request by ninjabear**

### CODE REVIEW

File: 3-enrich/scala-hadoop-enrich/src/test/scala/com.snowplowanalytics.snowplow.enrich.hadoop/good/CljTomcatSendgridEventSpec.scala [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+import JobSpecHelpers._
+
+/**
+  * Holds the input and expected data
+  * for the test.
+  */
+object CljTomcatSendgridEventSpec {
+
+  val lines = Lines(
+    "2014-10-09  16:28:31    -   13  255.255.255.255   POST    255.255.255.255   /com.sendgrid/v3   404 -  -    aid=email&cv=clj-0.6.0-tom-0.0.4&nuid=-   -   -   -   application%2Fjson   W3siZW1haWwiOiJleGFtcGxlQHRlc3QuY29tIiwidGltZXN0YW1wIjoxNDQ2NTQ5NjE1LCJzbXRwLWlkIjoiXHUwMDNjMTRjNWQ3NWNlOTMuZGZkLjY0YjQ2OUBpc210cGQtNTU1XHUwMDNlIiwiZXZlbnQiOiJwcm9jZXNzZWQiLCJjYXRlZ29yeSI6ImNhdCBmYWN0cyIsInNnX2V2ZW50X2lkIjoic1pST3dNR01hZ0Znbk9FbVNkdmhpZz09Iiwic2dfbWVzc2FnZV9pZCI6IjE0YzVkNzVjZTkzLmRmZC42NGI0NjkuZmlsdGVyMDAwMS4xNjY0OC41NTE1RTBCODguMCJ9XQ=="
```
Comment (ninjabear) [inline 3-enrich/scala-hadoop-enrich/src/test/scala/com.snowplowanalytics.snowplow.enrich.hadoop/good/CljTomcatSendgridEventSpec.scala]: Is `aid=email&cv=clj-0.6.0-tom-0.0.4&nuid=` a set of get params? Sendgrid doesn't appear to send any of these, but they should be captured still

Comment (jbeemster) [inline 3-enrich/scala-hadoop-enrich/src/test/scala/com.snowplowanalytics.snowplow.enrich.hadoop/good/CljTomcatSendgridEventSpec.scala]: They are indeed.  When you define a webhook endpoint in Mandrill or Mailchimp you could do something like this: `http://<your-collector>/com.sendgrid/v3?aid=email&cv=clj-0.6.0-tom-0.0.4

So every-time the webhook provider sends anything to us that query string will be sent with it.

Comment (ninjabear) [inline 3-enrich/scala-hadoop-enrich/src/test/scala/com.snowplowanalytics.snowplow.enrich.hadoop/good/CljTomcatSendgridEventSpec.scala]: Ok cool, thanks @jbeemster

File: 3-enrich/scala-hadoop-enrich/src/test/scala/com.snowplowanalytics.snowplow.enrich.hadoop/good/CljTomcatSendgridEventSpec.scala [hunk 2]
```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright (c) 2012-2014 Snowplow Analytics Ltd. All rights reserved.
```
Comment (jbeemster) [inline 3-enrich/scala-hadoop-enrich/src/test/scala/com.snowplowanalytics.snowplow.enrich.hadoop/good/CljTomcatSendgridEventSpec.scala]: The year needs to be bumped to 2015 again.

Comment (ninjabear) [inline 3-enrich/scala-hadoop-enrich/src/test/scala/com.snowplowanalytics.snowplow.enrich.hadoop/good/CljTomcatSendgridEventSpec.scala]: ah, sorry! Intellij hides that copyright notice at the top so I completely forgot about it. Fixed

**Comment by ninjabear** [comment]
@jbeemster I think that's everything?

**Comment by alexanderdean** [comment]
Wow! Looks great Ed...

**Comment by bernardosrulzon** [comment]
@ninjabear That's awesome, thanks so much! Quick question - does this PR also support the `unique_args` argument, used to pass custom parameters? That'd be super useful to measure the impact of specific campaigns, A/B Tests, etc.

**Comment by alexanderdean** [comment]
Hi @bernardosrulzon - no, this PR will silently drop those. I'll create a ticket to discuss this: #2202

**Comment by ninjabear** [comment]
I've updated the jsonpaths/ddl taking into account the jsonschema changes for https://github.com/snowplow/iglu-central/pull/277 here too. Hopefully that means this is ready to merge now, unless anyone has any other suggestions/comments!

**Pull request closed without merging**